code/game/objects/items/cards_ids.dm

```diff
@@ -871,6 +920,52 @@
 		return insert_money(interacting_with, user) ? ITEM_INTERACT_SUCCESS : ITEM_INTERACT_BLOCKING
 	return NONE
 
+/obj/item/card/id/item_ctrl_click(mob/user)
+	if(!trim)
+		balloon_alert(user, "card has no trim!")
+		return
+
+	if(!length(trim.honorifics))
+		balloon_alert(user, "card has no honorific to use!")
+		return
+
+	if(trim.honorific_positions == NONE)
+		balloon_alert(user, "no honorific positions! Error!")
+		stack_trace("ID card with honorifics found with no potential honorific positions!")
+		return
+
+	var/list/choices = list()
+	var/list/readable_names = HONORIFIC_POSITION_BITFIELDS()
+	for(var/i in readable_names) //Filter out the options you don't have on your ID.
+		if(trim.honorific_positions & readable_names[i]) //If the positions list has the same bit value as the readable list.
+			choices += i
+
+	var/chosen_position = tgui_input_list(user, "what position do you want your honorific in?", "Flair!", choices)
```

Comment:
You need to check after the input the user is still alive and in range of the id card and etc etc